cloudferry_devlab/cloudferry_devlab/generate_load.py

```diff
@@ -239,51 +240,74 @@ def chech_image_state_on_dst(self, img_id):
         img = self.dst_cloud.glanceclient.images.get(img_id)
         return img.status == 'active'
 
-    def update_filtering_file(self):
+    def set_tenant_filter(self, tenant, filter_dict, img_list, vm_list,
+                          not_incl_img, all_tenants_filter):
+        exclude_filter = tenant.get('exclude_images')
+        blacklisted_img = []
+        if exclude_filter:
+            blacklisted_img = [self.get_image_id(img)
+                               for img in self.config.images_blacklisted]
+        for key in filter_dict.keys():
+            if key == 'images':
+                for img_id in img_list:
+                    if not all_tenants_filter and exclude_filter:
+                        if img_id in blacklisted_img:
+                            filter_dict[key]['exclude_images_list'].append(
+                                    str(img_id))
+                    else:
+                        if img_id not in not_incl_img and \
+                                img_id not in blacklisted_img:
+                            if str(img_id) not in filter_dict[key][
+                                    'images_list']:
+                                filter_dict[key]['images_list'].append(
+                                    str(img_id))
```

Comment:
This code is not completely specific to blacklisted images. Its also a fix for [CF-456](https://mirantis.jira.com/browse/CF-456). This fix is related to filter only. If I move the blacklisted part out of this, it will be repetitive logic. I would say, we should not change it. I can add documentation for why is this logic added. @plomakin @roman-verchikov please suggest further.